<launch>
    <arg name="use_arduinos" default="true"/>

    <!--Robot model and tf transforms-->
    <arg name="model" default="$(find samana)/robot_description/urdf/samana.xml"/>
    <param name="robot_description" command="$(find xacro)/xacro $(arg model)" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true"/>

    <!--Data reading and decoding from Arduinos and control nodes-->
    <include file="$(find samana)/launch/arduino_start.launch">
        <arg name="use_arduinos" value="$(arg use_arduinos)"/>
    </include>

    <!--Robot localization-->
    <include file="$(find samana)/launch/localization.launch"/>

    <!--Mapping and path planning-->
    <include file="$(find samana)/launch/move_base.launch"/>
    <include file="$(find samana)/launch/follow_waypoints.launch"/>
    <node name="bug_algorithm" pkg="samana" type="bug_algorithm.py" respawn="true"/>

    <!--Object detection-->
    <node name="object_detector" pkg="samana" type="object_detector.py" respawn="true"/>

    <!--Launched separatly to start the run-->
    <!--<node name="main_state_machine" pkg="samana" type="main_smach.py" respawn="true"/>-->

    <!--Health check-->
    <node name="health_checker" pkg="samana" type="health_checker.py" respawn="true"/>
</launch>